plugins/restful/RestfulDataProviderVariable.php

```diff
@@ -0,0 +1,306 @@
+   *
+   * @return array
+   *   Array keyed by the sort field, with the order ('ASC' or 'DESC') as value.
+   */
+  public function defaultSortInfo() {
+    return array();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getVariablesForList() {
+    global $conf;
+    $variables = $conf;
+
+    $this->applyListSort($variables);
+    $this->applyListPagination($variables);
+
+    return $variables;
+  }
+
+  /**
+   * Sort the list of variables by name.
+   *
+   * @param string $a
+   *   The first variable name for the comparison.
+   * @param string $b
+   *   The second variable name for the comparison.
+   */
+  public function nameSort($a, $b) {
+    return strcasecmp($a, $b);
+  }
+
+  /**
+   * Sort the list of variables by value.
+   *
+   * If you are comparing variables with the same value, note that the PHP
+   * sorting algorithm is not stable, and their order may be changed.
+   *
+   * @param mixed $a
+   *   The first data structure for the comparison.
+   * @param mixed $b
+   *   The second data structure for the comparison.
+   */
+  public function valueSort($a, $b) {
+    return;
+  }
+
+  /**
+   * Sort the list of variables.
+   *
+   * This data provider does not handle compound sorts; the last sort defined
+   * will be the one to take effect.
+   *
+   * @param array $variables
+   *   An array keyed by variable name, valued by unserialized variable value.
+   *
+   * @throws \RestfulBadRequestException
+   */
+  protected function applyListSort(array &$variables) {
+    $public_fields = $this->getPublicFields();
+
+    // Get the sorting options from the request object.
+    $sorts = $this->parseRequestForListSort();
+
+    $sorts = $sorts ? $sorts : $this->defaultSortInfo();
+
+    foreach ($sorts as $public_field_name => $direction) {
+      if (isset($public_fields[$public_field_name]['property'])) {
+        $property_name = $public_fields[$public_field_name]['property'];
+        if ($property_name == 'name') {
+          uksort($variables, array(__CLASS__, 'nameSort'));
+        }
+        elseif ($property_name == 'value') {
+          uasort($variables, array(__CLASS__, 'valueSort'));
+        }
+      }
+      if ($direction == 'DESC') {
+        $variables = array_reverse($variables, TRUE);
+      }
+    }
+
+    return $variables;
+  }
+
+  /**
+   * Set correct page for the index within the array of variables.
+   *
+   * Determine the page that should be seen. Page 1 is actually index 0.
+   *
+   * @param array $variables
+   *   An array keyed by variable name, valued by unserialized variable value.
+   *
+   * @throws \RestfulBadRequestException
+   */
+  protected function applyListPagination(array &$variables) {
+    list($offset, $range) = $this->parseRequestForListPagination();
+    $variables = array_slice($variables, $offset, $range);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getTotalCount() {
+    return array('restful_count' => count($this->getVariablesForList()));
```

Comment:
The count should not include pagination. This should be something like `count($GLOBALS['conf'])`.

Also, I'm not sure I get why you are returning an array. This is what I see for other data providers:

```php
  /**
   * Helper method to get the total count of entities that match certain
   * request.
   *
   * @return int
   *   The total number of results without including pagination.
   */
  public function getTotalCount();
```